iib/workers/tasks/build.py

```diff
@@ -217,7 +222,7 @@ def _update_index_image_pull_spec(
     :raises IIBError: if the manifest list couldn't be created and pushed
     """
     conf = get_worker_config()
-    if from_index and overwrite_from_index:
+    if from_index and overwrite_from_index and resolved_prebuild_from_index:
```

Comment:
Is this logic change required?

Reply:
My bad, did not noticed it is checked in functions called after that.